fili-core/src/main/java/com/yahoo/bard/webservice/async/jobs/stores/ApiJobStoreFilter.java

```diff
@@ -0,0 +1,213 @@
+     * @param jobField  The JobField to perform the filtering on
+     * @param operation  The operation to perform (eg: eq)
+     * @param values  A Set of Strings to compare the JobField's value to.
+     */
+    private ApiJobStoreFilter(JobField jobField, ApiJobStoreFilterOperation operation, Set<String> values) {
+        this.jobField = jobField;
+        this.operation = operation;
+        this.values = values;
+    }
+
+    public JobField getJobField() {
+        return jobField;
+    }
+
+    public ApiJobStoreFilterOperation getOperation() {
+        return operation;
+    }
+
+    public Set<String> getValues() {
+        return values;
+    }
+
+    /**
+     * Construct an ApiJobStoreFilter object using the same ApiJobStoreFilterOperation and values as the object on
+     * which this method is called and using the supplied JobField.
+     *
+     * @param jobField  The JobField to perform the filtering on
+     *
+     * @return An instance of ApiJobStoreFilter created using the supplied JobField
+     */
+    public ApiJobStoreFilter withJobField(JobField jobField) {
+        return new ApiJobStoreFilter(jobField, operation, values);
+    }
+
+    /**
+     * Construct an ApiJobStoreFilter object using the same JobField and values as the object on
+     * which this method is called and using the supplied ApiJobStoreFilterOperation.
+     *
+     * @param operation  The operation to perform (eg: eq)
+     *
+     * @return An instance of ApiJobStoreFilter created using the supplied ApiJobStoreFilterOperation
+     */
+    public ApiJobStoreFilter withOperation(ApiJobStoreFilterOperation operation) {
+        return new ApiJobStoreFilter(jobField, operation, values);
+    }
+
+    /**
+     * Construct an ApiJobStoreFilter object using the same JobField and ApiJobStoreFilterOperation as the object on
+     * which this method is called and using the supplied values.
+     *
+     * @param values  A Set of Strings to compare the JobField's value to
+     *
+     * @return  An instance of ApiJobStoreFilter created using the supplied values
+     */
+    public ApiJobStoreFilter withValues(Set<String> values) {
+        return new ApiJobStoreFilter(jobField, operation, values);
+    }
+
+    /**
+     * Extracts the JobField to be examined from the tokenizedQuery.
+     *
+     * @param tokenizedQuery  The parsed ApiJobStore filter tokenizedQuery.
+     *
+     * @return  The JobField to be examined
+     * @throws BadFilterException is the JobField does not exist
+     */
+    private JobField extractJobField(Matcher tokenizedQuery) throws BadFilterException {
+        String fieldName = tokenizedQuery.group(1);
+        for (JobField field : DefaultJobField.values()) {
+            if (field.getName().equals(fieldName)) {
+                return field;
+            }
+        }
+        LOG.debug(FILTER_JOBFIELD_UNDEFINED.logFormat(fieldName));
+        throw new BadFilterException(FILTER_JOBFIELD_UNDEFINED.format(fieldName));
```

Comment:
This can be changed into a pretty little stream:

```java
return Arrays.stream(DefaultJobField.values())
                .filter(field -> field.getName().equals(fieldName))
                .findFirst()
                .orElseThrow(() -> {
                    LOG.debug(FILTER_JOBFIELD_UNDEFINED.logFormat(fieldName));
                    return new BadFilterException(FILTER_JOBFIELD_UNDEFINED.format(fieldName));
                });
```